extensions/wikia/AdEngine/js/AdInfoTracker.js

```diff
@@ -0,0 +1,95 @@
+/*global define*/
+define('ext.wikia.adEngine.adInfoTracker',  [
+	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.lookup.services',
+	'ext.wikia.aRecoveryEngine.recovery.helper',
+	'wikia.log',
+	'wikia.window'
+], function (adTracker, adContext, lookupServices, recoveryHelper, log, window) {
+	'use strict';
+
+	var	logGroup = 'ext.wikia.adEngine.adInfoTracker';
+
+	window.adEnginePvUID = window.adEnginePvUID || generateUUID();
+
+	function prepareData(slot, status) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)